test/functional/user_sessions_controller_test.rb

```diff
@@ -40,4 +40,58 @@ class UserSessionsControllerTest < ActionController::TestCase
       assert_equal I18n.t('user_sessions_controller.logged_in'), flash[:notice]
     end
   end
+
+  test 'sign up and login via provider basic flow' do
+    assert_not_nil OmniAuth.config.mock_auth[:google_oauth2]
+    #Omniauth hash is present
+    request.env['omniauth.auth'] =  OmniAuth.config.mock_auth[:google_oauth2]
+    assert_not_nil request.env['omniauth.auth']
+    #Sign Up for a new user
+    post :create
+    assert_equal "You have successfully signed in. Please change your password via a link sent to you via a mail",  flash[:notice]
+    #Log Out
+    post :destroy
+    assert_equal "Successfully logged out.",  flash[:notice]
+    #auth hash is present so login via a provider
+    post :create
+    assert_equal "Signed in!",  flash[:notice]
+  end
+
+  test 'sign up and login via provider alternative flow' do
+    assert_not_nil OmniAuth.config.mock_auth[:google_oauth2_2]
+    #Omniauth hash is present
+    request.env['omniauth.auth'] =  OmniAuth.config.mock_auth[:google_oauth2_2]
+    assert_not_nil request.env['omniauth.auth']
+    #Sign Up for an existing user as email exists in the db
+    post :create
+    assert_equal "Successfully linked to your account!",  flash[:notice]
+    #Log Out
+    post :destroy
+    assert_equal "Successfully logged out.",  flash[:notice]
+    #auth hash is present so login via a provider
+    post :create
+    assert_equal "Signed in!",  flash[:notice]
+  end
+
+
+  test 'login user with an email and then connect google provider' do
+    post :create, user_session: {
```

Comment:
Hi @SidharthBansal, can you change this passing of parameters like this `post :create, params: { user_session: {username: users(:jeff).email, password: 'secretive'} } as this is the new way of parameters in rails 5.0. There is also a deprecation warning related to this in the test.

Reply:
@Souravirus I have made the changes according to your suggestions. There is some places where we are logging in via provider instead of ``username`` and ``password`` so at those places I can't pass UserSession.

Reply:
Oh ok no need to change those where you are not passing any parameters.

Reply:
thanks

Reply:
Oh ok then let it be that way in rails 5.0 and we will see how to work around in rails 5.1. Thanks!!